internal/workflow/activities/unarchive.go

```diff
@@ -0,0 +1,178 @@
+package activities
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	"github.com/mholt/archiver/v3"
+
+	"github.com/artefactual-sdps/enduro/internal/fsutil"
+)
+
+const UnarchiveActivityName = "unarchive-activity"
+
+// UnarchiveActivity extracts transfer files from an archive (e.g. zip, tgz).
+type UnarchiveActivity struct {
+	logger logr.Logger
+}
+
+type UnarchiveParams struct {
+	// SourcePath is the path of the transfer to be unarchived.
+	SourcePath string
+
+	// StripTopLevelDir indicates whether the top-level "container" directory of
+	// the archive should be excluded from the destination directory.
+	StripTopLevelDir bool
+}
+
+type UnarchiveResult struct {
```

Comment:
It is standard to write NameActivityResult, NameActivityParams. This is done in all other activities as far as I am aware.